docs/HowTo/develop-subtopologies.md

```diff
@@ -0,0 +1,345 @@
+
+### MySubtopology.fpp
+
+As mentioned, this is a unified `.fpp` file. It not only includes the instances instance declarations for components that you'd like your subtopology to use, but also the definitions for the instances.
+
+Based on our example, we may have the following instance declarations:
+
+```cpp
+instance rng // RNG component instance
+instance rateGroup // rate group instance
+``` 
+
+and thus the following instance definitions:
+
+```cpp
+instance rng: MyLibrary.RNG base id 0xFF2FF \
+    queue size Defaults.QUEUE_SIZE \
+    stack size Defaults.STACK_SIZE \
+    priority 100
+
+instance rateGroup: Svc.ActiveRateGroup base id 0xFF4FF \
+    queue size Defaults.QUEUE_SIZE \
+    stack size DEFAULTS.STACK_SIZE \
+    priority 150
+
+instance rateGroupDriver: Svc.RateGroupDriver base id 0xFF8FF // to drive the 1Hz rate group
+```
+
+and also the following wiring:
+
+```cpp
+connections MyWiring {
+    // we will hook up the cycle for our rate group later on
+    rateGroup.RateGroupMemberOut[0] -> rng.run
+}
+```
+
+Thus, your unified fpp file, and so `MySubtopology.fpp` could look like:
+
+```cpp
+module MySubtopology {
+    instance rng: MyLibrary.RNG base id 0xFF2FF \
+        queue size Defaults.QUEUE_SIZE \
+        stack size Defaults.STACK_SIZE \
+        priority 100
+
+    instance rateGroup: Svc.ActiveRateGroup base id 0xFF4FF \
+        queue size Defaults.QUEUE_SIZE \
+        stack size DEFAULTS.STACK_SIZE \
+        priority 150
+
+    topology MySubtopology {
+        instance rng // RNG component instance
+        instance rateGroup // rate group instance
+
+        connections MyWiring {
+            rateGroup.RateGroupMemberOut[0] -> rng.run
+        }
+    } // end topology
+} // end MySubtopology
+```
+
+## MySubtopologyTopology.cpp and MySubtopologyTopology.hpp (pt1)
+
+This file is the bread and butter of developing your topology, as it runs any functions necessary to configure the topology, and also includes start and teardown functions for when the topology is started and after the topology is stopped respectively.
+
+As such, this file requires three implemented functions. The stubs of these functions should go in the `MySubtopologyTopology.hpp` file:
+
+```cpp
+namespace MySubtopology{
+    void configureTopology(const TopologyState& state) {
+        // ... your code here ...
+    }
+
+    void startTopology(const TopologyState& state) {
+        // ... your code here ...
+    }
+
+    void teardownTopology(const TopologyState& state) {
+        // ... your code here ...
+    }
+}
+```
+
+Additionally, the `hpp` file will need to include a file called `TopologyConfig.hpp`. The purpose of this is to include our main deployment's namespace, and to also include its autocoded `topology.cpp` file. Our subtopology will reference it, and will require access to the namespace *after* the autocoder has built it.
+
+In `fprime/config`, create a file called `TopologyConfig.hpp` and include:
```

Comment:
If there is multiple subtopologies, do they each include the same TopologyConfig.hpp?